presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -506,7 +506,6 @@ protected JdbcOutputTableHandle createTable(ConnectorSession session, ConnectorT
 
     protected String createTableSql(RemoteTableName remoteTableName, List<String> columns, ConnectorTableMetadata tableMetadata)
     {
-        checkArgument(tableMetadata.getProperties().isEmpty(), "Unsupported table properties: %s", tableMetadata.getProperties());
```

Comment:
undo

Reply:
There is no easy way to test Caching jdbc Client with different properties once this check is here.
There is still check in
```
Caused by: io.prestosql.spi.PrestoException: Catalog 'someCatalog' does not support table property 'someProperty'
	at io.prestosql.metadata.AbstractPropertyManager.getProperties(AbstractPropertyManager.java:96)
```

Reply:
As a rule of thumb, the need to test something should not cause the production code to have relaxed conditions.
I did not dive into this particular problem though, and I do not know how much effort we would need in these tests?

Reply:
You can modify `io.trino.plugin.jdbc.TestingH2JdbcClient` to implement custom properties. Or maybe you can inject different implementation of `BaseJdbcClient`. 

> As a rule of thumb, the need to test something should not cause the production code to have relaxed conditions.

Sometimes we have some extensions or introspection points added to the production code just for testing. See usages of `@VisibleForTesting`, but as @findepi mentioned this should not be the default way to go.

In this case this check is for a reason. So we could avoid mistakes and so bugs in production code.

Reply:
> Or maybe you can inject different implementation of BaseJdbcClient

This is something I want to avoid as it would require to override only one method (which will tend to be outdated) and delegate all others.

> As a rule of thumb, the need to test something should not cause the production code to have relaxed conditions.

Agree, but I'd say that this that exception when it's justified.

> In this case this check is for a reason. 

I'm fine with this check, because apart from 

> we could avoid mistakes and so bugs in production code.

it increases readability of this method. However it doubles logic of `AbstractPropertyManager.getProperties(AbstractPropertyManager.java:96)` and complicates testing.

> You can modify io.trino.plugin.jdbc.TestingH2JdbcClient to implement custom properties.

I'll take a look what I can do with this.

Reply:
> However it doubles logic of

It does not. The mentioned by you code verifies if you try to set not existing property. The one you removed is checking if properties are not ignored unnoticeably when doing CREATE TABLE.